flow/genericactors.actor.h

```diff
@@ -1227,7 +1226,7 @@ struct FlowLock : NonCopyable, public ReferenceCounted<FlowLock> {
 	// number of permits, and release() makes the caller no longer a holder of the lock. release() only runs waiting take()rs
 	// after the caller wait()s
 
-	struct Releaser : NonCopyable {
+	struct [[nodiscard]] Releaser : NonCopyable {
```

Comment:
I was hoping that this code would result in a warning after this change:

```
Releaser(lock); // Unintended, lock is released immediately
Releaser _(lock); // Intended
```

But that doesn't seem to be the case :/

Given that, I don't think there's much value in adding [[nodiscard]] here, so I think I'll remove it to avoid confusion.